lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1564,6 +1578,46 @@ static LogicalResult verifySubfieldOp(SubfieldOp op) {
   return success();
 }
 
+/// Check that a value is driven by a constant value. Looks through subindex and
+/// subfield ops.
+static bool isDrivenByConstant(Value op) {
+  for (auto &use : op.getUses()) {
+    auto result = TypeSwitch<Operation *, bool>(use.getOwner())
+                      .Case<SubindexOp, SubfieldOp>([](auto op) {
+                        return isDrivenByConstant(op.getResult());
+                      })
+                      .Case<SubaccessOp>([](auto) { return false; })
+                      .Case<PartialConnectOp, ConnectOp>([&](auto op) {
+                        if (use.get() == op.dest())
+                          return isConstant(op.src());
+                        else
+                          return true;
+                      })
+                      .Default([](auto) { return true; });
+    if (!result)
+      return false;
+  }
+  return true;
+}
```

Comment:
This will also need to look through casts.  Chisel likes to emit constants like: `asSInt(UInt<1>(0))` which can cause issues here. 

This test is currently failing with this PR:

```scala
circuit Foo:
  module Foo:
    input clock : Clock
    input reset : AsyncReset
    input x : SInt<4>
    output y : SInt<4>
    output z : SInt<4>
    reg r : SInt<4>, clock with : (reset => (reset, asSInt(UInt(0))))
    r <= x
    wire w : SInt<4>
    reg r2 : SInt<4>, clock with : (reset => (reset, w))
    r2 <= x
    node n = UInt("hf")
    w <= asSInt(n)
    y <= r2
    z <= r
```

This was extracted from [`AsyncResetSpec.scala`](https://github.com/chipsalliance/firrtl/blob/master/src/test/scala/firrtlTests/AsyncResetSpec.scala#L235).  You may want to also check the tests in there about complex literals and bundle literals.

Reply:
I was staring at that function thinking "I'm sure I missed some obvious case". Here we go. Also completely forgot about the `AsyncResetSpec`.